integrated_channels/moodle/transmitters/content_metadata.py

```diff
@@ -21,6 +34,105 @@ def __init__(self, enterprise_configuration, client=MoodleAPIClient):
+                    # we just need to record the exception and give up.
+                    sleep(5)
+                    count += 1
+                    self._transmit_create(channel_metadata_item_map, chunk_size=LOW_CHUNK_SIZE, count=count)
+                else:
+                    LOGGER.error(
+                        'Failed to create [%s] content metadata items for integrated channel [%s] [%s]. '
+                        'Task failed with message [%s] and status code [%s]',
+                        len(chunk),
+                        self.enterprise_configuration.enterprise_customer.name,
+                        self.enterprise_configuration.channel_code(),
+                        exc.message,
+                        exc.status_code
+                    )
+                    LOGGER.exception(exc)
+            else:
+                self._create_transmissions(chunk)
+
+    def _transmit_update(self, channel_metadata_item_map, transmission_map, chunk_size=None, count=0):  # pylint: disable=arguments-differ
+        """
+        Transmit content metadata update to integrated channel.
+        """
+        chunk_size = chunk_size if chunk_size else self.enterprise_configuration.transmission_chunk_size
+        chunk_items = chunks(channel_metadata_item_map, chunk_size)
+        transmission_limit = settings.INTEGRATED_CHANNELS_API_CHUNK_TRANSMISSION_LIMIT.get(
+            self.enterprise_configuration.channel_code()
+        )
+        for chunk in islice(chunk_items, transmission_limit):
+            serialized_chunk = self._serialize_items(list(chunk.values()))
+            try:
+                self.client.update_content_metadata(serialized_chunk)
+            except ClientError as exc:
+                if exc.status_code == 414 and count <= 1:
+                    count += 1
+                    sleep(5)
+                    self._transmit_update(channel_metadata_item_map, transmission_map, chunk_size=LOW_CHUNK_SIZE, count=count)
+                else:
+                    LOGGER.error(
+                        'Failed to update [%s] content metadata items for integrated channel [%s] [%s]. '
+                        'Task failed with message [%s] and status code [%s]',
+                        len(chunk),
+                        self.enterprise_configuration.enterprise_customer.name,
+                        self.enterprise_configuration.channel_code(),
+                        exc.message,
+                        exc.status_code
+                    )
+                    LOGGER.exception(exc)
+            else:
+                self._update_transmissions(chunk, transmission_map)
+
+    def _transmit_delete(self, channel_metadata_item_map, chunk_size=None, count=0):  # pylint: disable=arguments-differ
+        """
+        Transmit content metadata deletion to integrated channel.
+        """
+        chunk_size = chunk_size if chunk_size else self.enterprise_configuration.transmission_chunk_size
```

Comment:
I feel like since this snippet is now used several places including base transmitter, does it makes sense now to make it a method in the base class and re use it here?

Reply:
Not sure I grasp. This chunk size definition is not used in the base class. I added the chunk_size parameter and thus needed to account for it here.

Reply:
oh not sure if I misread I thought the few lines that produce the tuple values were the same in all the methods, but if the base does not also have the same code lines probably not worth it. was looking for avoidance of dup if possible, if anything you can make it a shared method in this very class instead too if you like

Reply:
I think the base integrated channels has a util function `chunks` method we call later that also does islicing and I didn't dig into deeply to understand why we seem to duplicate that even in the base transmitter, but other than that no, no duplication really here except obviously defining this over and over for the 3 overrides.

yeah I had thought about just having an entire wrapper honestly, since ultimately all of these `_transmit_*` overrides are just: use chunk size, attempt a transmission, recurse on 414. But the wrapping inception and recursiveness seemed gross together and a nightmare to eventually test.